src/components/Tool/FixCommitmentInfo/Contact.tsx

```diff
@@ -249,9 +249,10 @@ const Contact: React.FC<Props> = ({
         initialValues={{
           statusValue: statusValue,
           pledgeCurrency: amountCurrency,
-          pledgeAmount: amount || suggestedAmount,
```

Comment:
@canac Can you help us make sense of this? We don't know if Fix Commitment Info should set the initial values to blank, the suggested values, or the current values.  On my personal account, it appears that it sets the initial values to the suggested values and leaves it blank if there is no suggested value.

However, in the Angular code, the `mutateContacts` service appears to use `defaultTo` to set the default to the original/current value and only use the suggested value if it doesn't have one. I was thinking you might be able to understand the angular better than we can.

https://github.com/CruGlobal/mpdx_web/blob/19c03ae0689609eeee8b9b516700cc75ee4f3f79/src/tools/fix/commitment/commitment.service.ts

Reply:
On your account, is there a way to log out the `contact.pledge_amount`? I wonder if this is blank, which is why it's showing blank.

When you say blank, do you mean like this? where it only shows the "0" placeholder for the amount.
<img width="940" alt="Screenshot 2024-09-16 at 3 48 39 PM" src="https://github.com/user-attachments/assets/ddc2f3c2-24dd-4ca5-8e10-2b88a0a53237">

Reply:
If so, this is because the pledge amount is `null` and there is no `contact.suggested_changes.pledge_amount`
<img width="490" alt="Screenshot 2024-09-16 at 3 49 45 PM" src="https://github.com/user-attachments/assets/4cdd0f0d-e258-43bd-a13e-5dd16058d446">

Reply:
Let me know if that makes sense

Reply:
Here's an example. Clearly, on this contact there is a current `status` and pledge `frequency`. But they are not set as the initial values in the form fields. I think this is the correct behavior, but I'm not seeing how that happens in the code.

<img width="981" alt="Screenshot 2024-09-16 at 1 12 34 PM" src="https://github.com/user-attachments/assets/58d68c2b-345f-48ed-b6e5-36ec4131e8b3">

Reply:
Ok, it looks like the data from angular is messed up...
``            "attributes": {
                "avatar": "https://stage.mpdx.org/images/avatar_f.png",
                "name": "Dauterman, Abigail",
                "pledge_amount": "150.0",
                "pledge_currency": "USD",
                "pledge_frequency": "6",
                "status": "partner_financial",
                "suggested_changes": {
                    "pledge_amount": null,
                    "pledge_frequency": null,
                    "status": "Partner - Special"
                }``

Reply:
The above was staging. So maybe because the `status` data was wrong it was showing up blank.
But here is production. It sets the initial value to `suggested_changes` `status` instead of the current `status`. What do you think is going on here?

<img width="386" alt="Screenshot 2024-09-16 at 1 46 03 PM" src="https://github.com/user-attachments/assets/9347cf6e-ae4a-4ba2-bd71-432b40fc3da1">
<img width="981" alt="Screenshot 2024-09-16 at 1 46 14 PM" src="https://github.com/user-attachments/assets/37c631fd-d9a1-449f-8c20-9c2f3885cc71">

Reply:
Is it possible that the discrepancy is because task phases is in staging but not production? Maybe in staging it tried to set the autocomplete value to `Partner - Special` but there's not an item with that value so it shows up as blank?

Reply:
Yes I think that is what is happening in staging. I think that is explainable.

But for production (most recent screenshots) that is not the case. It is initializing the suggested_changes status.

Reply:
Interesting. From what I'm looking at in the code, I do not think it's initializing the `suggested_changes` status as you haven't hooked up the `suggested_changes.status` to the initial value of `status`.

Reply:
mpdx-react/src/components/Tool/FixCommitmentInfo/Contact.tsx
```
const Contact: React.FC<Props> = ({
  statusTitle,
  statusValue,
  ...
  suggestedChanges,
}) => {
<Formik
        initialValues={{
          statusValue: statusValue,
          ...
          }}
/>
          
```


mpdx-react/src/components/Tool/FixCommitmentInfo/FixCommitmentInfo.tsx
```
<Contact
                        statusTitle={
                          contact.status &&
                          contactPartnershipStatus[contact.status]
                        }
                        statusValue={
                          getLocalizedContactStatus(t, contact.status) || ''
                        }
                        statuses={contactStatuses || []}
                        suggestedChanges={formatSuggestedChanges(
                          contact?.suggestedChanges,
                        )}
 />
```

Reply:
The only thing I can think of is the `contactStatuses` variable is incorrect, but as you said it shouldn't be.

Reply:
@dr-bizz my comments above are about angular, not React. We are trying to figure out what the desired behavior is in angular. Then we can replicate it in React. Right now it is unclear what the angular code is doing.

Reply:
src/tools/fix/commitment/commitment.service.ts
```
assign(contact, {
    ....
    original_status: contact.status,
    ...
    status: defaultTo(contact.status, contact.suggested_changes.status),
    ...
}),
```

src/tools/fix/commitment/item/item.html
```
<div class="contact">
                <a ng-href="/contacts/{{$ctrl.contact.id}}" target="_blank"
                    ><h4 class="name">{{ $ctrl.contact.name }}</h4></a
                >
                ...
                {{ $ctrl.contact.original_status }} {{ $ctrl.contact.original_pledge_amount | currency : '' }} {{
                $ctrl.contact.original_pledge_currency }} {{ $ctrl.contact.original_pledge_frequency }}
</div>

...

<div class="status">
                    <select
                        chosen
                        required
                        disable-search="true"
                        data-placeholder-text-single="'Status' | translate"
                        class="form-control"
                        id="status"
                        ng-options="opt.id as opt.value for opt in $ctrl.serverConstants.data.status_hashes"
                        ng-model="$ctrl.contact.status"
                    >
                        <option value=""></option>
                    </select>
</div>
```

Reply:
Does that make sense?

Reply:
status: suggestedChanges.status || statusValue

Reply:
Yes, that is correct

Reply:
Nice. So we have confirmed that the suggested values (status, frequency, amount) should be the initial form values. If they don't exist, then it will set the current values as the initial form values.